cmd/piper.go

```diff
@@ -82,13 +82,17 @@ func PrepareConfig(cmd *cobra.Command, metadata *config.StepData, stepName strin
 		var customConfig io.ReadCloser
 		var err error
 		//accept that config file and defaults cannot be loaded since both are not mandatory here
-		if piperutils.FileExists(GeneralConfig.CustomConfig) {
-			if customConfig, err = openFile(GeneralConfig.CustomConfig); err != nil {
-				errors.Wrapf(err, "Cannot read '%s'", GeneralConfig.CustomConfig)
-			}
+		if projectConfigFile, err := getProjectConfigFile(GeneralConfig.CustomConfig); err != nil {
+			return err
 		} else {
```

Comment:
you don't need the else here. This should also make the error go away which you were wondering about.

Reply:
That I already understood. But  I think it is handy to have `err` with the current limited scope. But having it with this scope results in a CC-message. Basically the question is: what is bad with an instant `return`?

Reply:
maybe you want to reconsider keeping err in the limited scope to solve the issue ...